src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
I also like the idea of using seek(). I was talking to @cesarbs, but forgot that Unescape modifies the blocks, so I forgot it could work for both utf8 and ascii strings assuming `NormalizeToNFC` doesn't convert anything to a `.` and `\`.

Considering we're only taking a ~1% perf hit for the plaintext benchmark according to @cesarbs testing, I think that this can go in as is. But if @cesarbs has time to change it to use seek, I more than welcome it.